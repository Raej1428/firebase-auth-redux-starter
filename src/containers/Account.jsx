// @flow
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { updateUserRecord, updatePassword } from '../actions/auth';
import Account from '../components/Account';
import { addSuccess, addError } from '../actions/toast';

const mapStateToProps = state => ({
  isAuthenticated: state.auth.authenticated,
  profile: state.auth.profile,
  uid: state.auth.uid,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: (uid, data) => updateUserRecord(uid, data),
  updatePassword: newPassword => updatePassword(newPassword),
  sendSuccessMessage: message => dispatch(addSuccess(message)),
  sendErrorMessage: message => dispatch(addError(message)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Account));
