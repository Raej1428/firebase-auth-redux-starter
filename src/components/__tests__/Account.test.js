/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import Account from '../Account';
import ChangePasswordForm from '../ChangePasswordForm'
import UpdateProfileForm from '../UpdateProfileForm'

describe('Account component', () => {
  it('root account component matches snapshot ', () => {
    const wrapper = shallow(
      <Account />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('changePasswordForm component matches snapshot ', () => {
    const wrapper = shallow(
      <ChangePasswordForm />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('UpdateProfileForm component matches snapshot ', () => {
    const wrapper = shallow(
      <UpdateProfileForm profile={{ firstName: "Test", lastName: "Guy" }} />
    );
    expect(wrapper).toMatchSnapshot();
  })
})
