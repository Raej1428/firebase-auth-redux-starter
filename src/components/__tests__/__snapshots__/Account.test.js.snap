// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component UpdateProfileForm component matches snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <C
    profile={
        Object {
            "firstName": "Test",
            "lastName": "Guy",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": true,
      "handleSubmit": [Function],
      "initialValues": Object {
        "firstName": "Test",
        "lastName": "Guy",
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "profile": Object {
        "firstName": "Test",
        "lastName": "Guy",
      },
      "render": [Function],
      "validate": undefined,
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": true,
        "handleSubmit": [Function],
        "initialValues": Object {
          "firstName": "Test",
          "lastName": "Guy",
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "profile": Object {
          "firstName": "Test",
          "lastName": "Guy",
        },
        "render": [Function],
        "validate": undefined,
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Account component changePasswordForm component matches snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <C />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": true,
      "handleSubmit": [Function],
      "initialValues": Object {
        "password": "",
        "passwordConfirm": "",
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "render": [Function],
      "validate": undefined,
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": true,
        "handleSubmit": [Function],
        "initialValues": Object {
          "password": "",
          "passwordConfirm": "",
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "render": [Function],
        "validate": undefined,
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Account component root account component matches snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Account />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f2 lh-title mh3"
>
          Hi from account
</h1>,
        <div
          className="cf ph2-ns"
>
          <div
                    className="fl w-100 w-50-ns pa2"
          >
                    <div
                              className=" bg-white pa2"
                    >
                              <C
                                        profile={undefined}
                                        sendSuccessMessage={undefined}
                                        uid={undefined}
                                        updateProfile={undefined}
                              />
                    </div>
          </div>
          <div
                    className="fl w-100 w-50-ns pa2"
          >
                    <div
                              className=" bg-white pa2"
                    >
                              <C
                                        sendErrorMessage={undefined}
                                        sendSuccessMessage={undefined}
                                        updatePassword={undefined}
                              />
                    </div>
          </div>
</div>,
      ],
      "className": "mw10 center ph3-ns",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hi from account",
          "className": "f2 lh-title mh3",
        },
        "ref": null,
        "rendered": "Hi from account",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fl w-100 w-50-ns pa2"
>
              <div
                            className=" bg-white pa2"
              >
                            <C
                                          profile={undefined}
                                          sendSuccessMessage={undefined}
                                          uid={undefined}
                                          updateProfile={undefined}
                            />
              </div>
</div>,
            <div
              className="fl w-100 w-50-ns pa2"
>
              <div
                            className=" bg-white pa2"
              >
                            <C
                                          sendErrorMessage={undefined}
                                          sendSuccessMessage={undefined}
                                          updatePassword={undefined}
                            />
              </div>
</div>,
          ],
          "className": "cf ph2-ns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className=" bg-white pa2"
>
                <C
                                profile={undefined}
                                sendSuccessMessage={undefined}
                                uid={undefined}
                                updateProfile={undefined}
                />
</div>,
              "className": "fl w-100 w-50-ns pa2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <C
                  profile={undefined}
                  sendSuccessMessage={undefined}
                  uid={undefined}
                  updateProfile={undefined}
/>,
                "className": " bg-white pa2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "profile": undefined,
                  "sendSuccessMessage": undefined,
                  "uid": undefined,
                  "updateProfile": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className=" bg-white pa2"
>
                <C
                                sendErrorMessage={undefined}
                                sendSuccessMessage={undefined}
                                updatePassword={undefined}
                />
</div>,
              "className": "fl w-100 w-50-ns pa2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <C
                  sendErrorMessage={undefined}
                  sendSuccessMessage={undefined}
                  updatePassword={undefined}
/>,
                "className": " bg-white pa2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "sendErrorMessage": undefined,
                  "sendSuccessMessage": undefined,
                  "updatePassword": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f2 lh-title mh3"
>
            Hi from account
</h1>,
          <div
            className="cf ph2-ns"
>
            <div
                        className="fl w-100 w-50-ns pa2"
            >
                        <div
                                    className=" bg-white pa2"
                        >
                                    <C
                                                profile={undefined}
                                                sendSuccessMessage={undefined}
                                                uid={undefined}
                                                updateProfile={undefined}
                                    />
                        </div>
            </div>
            <div
                        className="fl w-100 w-50-ns pa2"
            >
                        <div
                                    className=" bg-white pa2"
                        >
                                    <C
                                                sendErrorMessage={undefined}
                                                sendSuccessMessage={undefined}
                                                updatePassword={undefined}
                                    />
                        </div>
            </div>
</div>,
        ],
        "className": "mw10 center ph3-ns",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hi from account",
            "className": "f2 lh-title mh3",
          },
          "ref": null,
          "rendered": "Hi from account",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fl w-100 w-50-ns pa2"
>
                <div
                                className=" bg-white pa2"
                >
                                <C
                                                profile={undefined}
                                                sendSuccessMessage={undefined}
                                                uid={undefined}
                                                updateProfile={undefined}
                                />
                </div>
</div>,
              <div
                className="fl w-100 w-50-ns pa2"
>
                <div
                                className=" bg-white pa2"
                >
                                <C
                                                sendErrorMessage={undefined}
                                                sendSuccessMessage={undefined}
                                                updatePassword={undefined}
                                />
                </div>
</div>,
            ],
            "className": "cf ph2-ns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className=" bg-white pa2"
>
                  <C
                                    profile={undefined}
                                    sendSuccessMessage={undefined}
                                    uid={undefined}
                                    updateProfile={undefined}
                  />
</div>,
                "className": "fl w-100 w-50-ns pa2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <C
                    profile={undefined}
                    sendSuccessMessage={undefined}
                    uid={undefined}
                    updateProfile={undefined}
/>,
                  "className": " bg-white pa2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "profile": undefined,
                    "sendSuccessMessage": undefined,
                    "uid": undefined,
                    "updateProfile": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className=" bg-white pa2"
>
                  <C
                                    sendErrorMessage={undefined}
                                    sendSuccessMessage={undefined}
                                    updatePassword={undefined}
                  />
</div>,
                "className": "fl w-100 w-50-ns pa2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <C
                    sendErrorMessage={undefined}
                    sendSuccessMessage={undefined}
                    updatePassword={undefined}
/>,
                  "className": " bg-white pa2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "sendErrorMessage": undefined,
                    "sendSuccessMessage": undefined,
                    "updatePassword": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
